x-pack/solutions/search/plugins/enterprise_search/server/deprecations/routes.test.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { mockDependencies, MockRouter } from '../__mocks__';
+
+import { registerDeprecationRoutes } from './routes';
+
+describe('deprecation routes', () => {
+  describe('POST /internal/enterprise_search/deprecations/delete_crawler_connectors', () => {
+    let mockRouter: MockRouter;
+
+    beforeEach(() => {
+      jest.clearAllMocks();
+      mockRouter = new MockRouter({
+        method: 'post',
+        path: '/internal/enterprise_search/deprecations/delete_crawler_connectors',
+      });
+
+      registerDeprecationRoutes({
+        ...mockDependencies,
+        router: mockRouter.router,
+      });
+    });
+
+    it('validates correctly with ids and deprecation context', () => {
+      const request = {
+        body: { ids: ['foo'], deprecationDetails: { domainId: 'enterpriseSearch' } },
+      };
+      mockRouter.shouldValidate(request);
+    });
+
+    it('fails validation without ids', () => {
+      const request = { body: { deprecationDetails: { domainId: 'enterpriseSearch' } } };
+      mockRouter.shouldThrow(request);
+    });
+
+    it('fails validation without deprecation context', () => {
+      const request = { body: { ids: ['foo'] } };
+      mockRouter.shouldThrow(request);
+    });
+  });
+
+  describe('POST /internal/enterprise_search/deprecations/convert_connectors_to_client', () => {
```

Comment:
Similar here could we add a test for the happy path of `putUpdateNative` and if `putUpdateNative` promise result rejects.